packages/twenty-server/src/engine/metadata-modules/view-group/resolvers/view-group.resolver.ts

```diff
@@ -49,11 +56,28 @@ export class ViewGroupResolver {
   }
 
   @Mutation(() => ViewGroupDTO)
-  @UseGuards(SettingsPermissionsGuard(PermissionFlagType.VIEWS))
+  @UseGuards(CustomPermissionGuard)
   async createCoreViewGroup(
     @Args('input') createViewGroupInput: CreateViewGroupInput,
     @AuthWorkspace() { id: workspaceId }: WorkspaceEntity,
+    @AuthUserWorkspaceId() userWorkspaceId: string | undefined,
   ): Promise<ViewGroupDTO> {
+    const view = await this.viewRepository.findOne({
+      where: {
+        id: createViewGroupInput.viewId,
+        workspaceId,
+        deletedAt: IsNull(),
+      },
+    });
+
+    if (!isDefined(view)) {
+      throw new Error('View not found');
+    }
+
+    if (view.createdById !== userWorkspaceId) {
+      throw new Error('You do not have permission to update this view');
```

Comment:
we should throw gql errors here (or CustomExceptions and add an exception handler)